### ISSUE TITLE
Incorrect contours in `contourf[!]` with `PolarAxis` using GLMakie

### ISSUE THREAD

**Issue by anowacki** [MakieOrg/Makie.jl#3410]
When plotting a filled contour plot with `contourf!` into a `PolarAxis`, the filled contours are sometimes incorrect when using GLMakie.  `contour!` is fine, as is `surface!`.  (~`tricontourf!` is okay but not as (correctly) smooth as `contour!`, which is an understandable algorithmic reason and not a bug.~ Edit: `tricontourf!` actually has the same problem as the image shows.)

For example, plotting black contours from `contour!` on top of each kind of plot, with black dots where the data are, we can see the boundaries between the filled colours for the `contourf!` plot do not match and are not as they should be:

![funny_contours_GLMakie](https://github.com/MakieOrg/Makie.jl/assets/1231209/aa2127f0-804a-401a-b126-e5bcbb6591dd)

CairoMakie works fine, however:

![normal_contours](https://github.com/MakieOrg/Makie.jl/assets/1231209/2ec85d5a-28f1-46e9-bba8-29b47e123e4d)

I have tried passing in the arguments as matrices of scattered points, passing just vectors for the coordinates, and removing the points at `r = 0` in case that was confusing things, all to no avail.

The following reproducer shows this:

```julia
using GLMakie

az = 0:20:360 # Azimuths in °
inc = 0:10:90 # Inclinations in °
# Convert to polar plot coordinates
θ = deg2rad.(az)
r = cosd.(inc) # Or just (90 .- inc)./90
# Data
z = [
     9773.9   9703.68  9502.15  9198.09  8843.41  8512.5   8278.24  8161.89  8124.4   8118.1
     9500.59  9440.42  9269.0   9017.47  8736.85  8489.3   8318.63  8221.81  8164.38  8118.1
     8811.91  8787.13  8704.11  8590.19  8477.58  8388.27  8322.44  8264.1   8197.3   8118.1
     8111.8   8152.07  8194.78  8243.18  8291.12  8324.74  8329.08  8294.56  8220.84  8118.1
     7793.81  7885.61  8002.2   8127.17  8238.17  8314.1   8340.5   8312.01  8233.2   8118.1
     7793.81  7885.61  8002.2   8127.17  8238.17  8314.1   8340.5   8312.01  8233.2   8118.1
     8111.8   8152.07  8194.78  8243.18  8291.12  8324.74  8329.08  8294.56  8220.84  8118.1
     8811.91  8787.13  8704.11  8590.19  8477.58  8388.27  8322.44  8264.1   8197.3   8118.1
     9500.59  9440.42  9269.0   9017.47  8736.85  8489.3   8318.63  8221.81  8164.38  8118.1
     9773.9   9703.68  9502.15  9198.09  8843.41  8512.5   8278.24  8161.89  8124.4   8118.1
     9500.59  9437.53  9257.28  8984.82  8666.76  8370.94  8166.83  8080.84  8080.87  8118.1
     8811.91  8763.95  8644.13  8469.66  8274.38  8104.89  8004.77  7989.61  8039.26  8118.1
     8111.8   8066.22  8010.23  7946.45  7886.85  7850.47  7856.04  7911.26  8006.3   8118.1
     7793.81  7732.23  7697.15  7682.61  7686.87  7714.51  7773.31  7866.92  7988.07  8118.1
     7793.81  7732.23  7697.15  7682.61  7686.87  7714.51  7773.31  7866.92  7988.07  8118.1
     8111.8   8066.22  8010.23  7946.45  7886.85  7850.47  7856.04  7911.26  8006.3   8118.1
     8811.91  8763.95  8644.13  8469.66  8274.38  8104.89  8004.77  7989.61  8039.26  8118.1
     9500.59  9437.53  9257.28  8984.82  8666.76  8370.94  8166.83  8080.84  8080.87  8118.1
     9773.9   9703.68  9502.15  9198.09  8843.41  8512.5   8278.24  8161.89  8124.4   8118.1
]

# For use with `tricontourf!`
az_mat = repeat(az, 1, length(inc))
inc_mat = repeat(inc', length(az))
θ_mat = deg2rad.(az_mat)
r_mat = cosd.(inc_mat)

nrows = 2

fig = Figure(size=(800,800))

for (i, func!) in enumerate((surface!, contour!, tricontourf!, contourf!,))
    irow = mod1(i, nrows)
    icol = fld1(i, nrows)
    ax = PolarAxis(fig[irow,icol];
        rlimits=(0, 1),
        thetaticklabelsvisible=false,
        rticklabelsvisible=false,
        thetagridvisible=false,
        rgridvisible=false,
        title=string(func!)
    )

    args = if func! == tricontourf!
        vec(θ_mat), vec(r_mat), vec(z)
    else
        θ, r, z
    end

    levels = range(extrema(z)..., length=20)

    func!(ax, args...; colormap=:turbo, shading=NoShading, linewidth=3)

    contour!(ax, θ, r, z; overdraw=true, color=:black, linewidth=1, levels)

    scatter!(ax, vec(θ_mat), vec(r_mat), color=:black, markersize=5, overdraw=true)
end

fig |> display
```

(There is also another bug: GLMakie doesn't draw the contours on top of the surface plot, despite calling `overdraw=true` for `contour!`.  Will open another issue for that.)

- [x] are you running newest version (version from docs) ?
  - GLMakie v0.9.0
  - CairoMakie v0.11.0
  - Makie v0.20.0
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [x] What platform + GPU are you on?
  - macOS 10.15.7 and 12.7
  - AMD FirePro D300 2 GB

**Comment by ffreyer** [comment]
Probably needs to resolve the transformation before running the contour algorithm similar to voronoiplots. I.e.
https://github.com/MakieOrg/Makie.jl/blob/822d11cf1e9751084a276f79b616b03afdc43e2d/src/basic_recipes/voronoiplot.jl#L146-L149
with the transform function getting overwritten for lower level plot calls
https://github.com/MakieOrg/Makie.jl/blob/822d11cf1e9751084a276f79b616b03afdc43e2d/src/basic_recipes/voronoiplot.jl#L166-L167
and custom `data_limits` to avoid them becoming untransformed limits
https://github.com/MakieOrg/Makie.jl/blob/822d11cf1e9751084a276f79b616b03afdc43e2d/src/basic_recipes/voronoiplot.jl#L170-L181

**Comment by ffreyer** [comment]
Seems like this got fixed, maybe by #3402?

With v0.20.1 I get:

![Screenshot from 2023-11-28 18-02-04](https://github.com/MakieOrg/Makie.jl/assets/10947937/9f6b713c-cea6-4a4b-b997-d4aaa185962b)

**Comment by anowacki** [comment]
Should have waited a day before submitting the issue—thanks!
